### ISSUE TITLE
Tilde completion has been buggy since bash 5.x.x

### ISSUE THREAD

**Issue by dsifford** [scop/bash-completion#298]
Hello there.

I've been suffering with this for a while now and I finally got fed up enough to write out this issue.

Ever since (ostensibly) the release of bash 5, tilde completion has been doing some extremely aggravating things.

### Here's a concrete example:

I'm currently in a directory inside my home directory, `~/repos/foo`. Here's the contents of that directory...

```
~/repos/foo $ tree
.
|-- bar
|-- baz
`-- foo

0 directories, 3 files
```

Now, for whatever reason, when I try to do a completion with, say `bind` as an easy example, the following happens...

```
~/repos/foo $ bind -f ~/.confi|<TAB>
```

becomes...

```
~/repos/foo $ bind -f bar |
```

Where `|` is my cursor.

The same is true for any literally typed, valid directory chain beginning with `~`.

Any idea what could be going on here?

Thanks in advance :pray:

**Comment by scop** [comment]
I'm not using bash 5 myself yet, so no specific ideas. But there are the general troubleshooting instructions, see "Troubleshooting" in README.md. You don't mention the bash-completion version; be sure to try out with the latest from master branch here.

**Comment by dsifford** [comment]
Hey there.

Sorry for the lack of info. I went through the readme stuff pretty thoroughly to no avail. I'm on arch linux, so both bash and bash-completion are the most current...

- Bash 5.0.2(1)-release
- bash-completion 2.8-1 

I also notice this on my macbook, running the latest bash and bash-completion.

**Comment by scop** [comment]
I'll try to have a look at this somehow. But note that while bash-completion 2.8 is the latest "released" version at the moment, it's over a year old and there have been almost 400 commits since (not all of them are actual code changes though), that's why I asked to test with the latest from the master branch here...

**Comment by scop** [comment]
I gave this a quick try in a Fedora development docker image which has bash 5.0.2 and with which we run our automated test suite -- could not reproduce using your recipe.

I suggest trying out with as minimal environment as possible, for example from a `bash --norc` session (with bash_completion not loaded nor any of your possible configs, moving them incl e.g. ~/.inputrc in addition to bash config files away temporariliy), then trying to find out what is the thing that makes the incorrect behavior appear.

Also, note that in the troubleshooting instructions you mentioned having gone through is a note about the debugging output and how it helps us, but obviously it won't do that if we never see it... so attaching it here (after looking that it doesn't contain anything you don't want to share) might help.

**Comment by dsifford** [comment]
Hm, shoot.. I'll have to dig into this a bit more when I get some time and report back. 

In the meantime, if you're bored/curious, here's my inputrc (which seems like it might be the only variable between your fresh machine test and mine)...

https://github.com/dsifford/.dotfiles/blob/master/config/readline/inputrc

**Comment by dsifford** [comment]
Also, of note... The command in my example matters... 

If, for example, I use the `ssh` command, this doesn't appear to happen...

So `ssh ~/some/valid/file` works and completes without any issue, `bind -f ~/some/valid/file` does not

**Comment by dsifford** [comment]
Looks like this is fixed in 2.9.

I'll reopen if something turns up otherwise.
